tracker/sm-website-update-visitor.js

```diff
@@ -1,147 +1,153 @@
+(function(sm) {
+
+  updateVisitorCustomAttributes = function(attributeName, attributeValue){
+    getCurrentVisitor(salemove).then(function(visitor) {
+      visitor.custom_attributes[attributeName] = attributeValue
+      $.ajax({
+        type: "POST",
+        beforeSend: function (request)
+          {
+            requestHeaders = sm.salemoveApi.getRequestHeaders();
+              Object.keys(requestHeaders).forEach(function(key) {
+              request.setRequestHeader(key, requestHeaders[key]);
+            });
+          },
+        url: "https://api.salemove.com/visitor",
+        data:
+          {
+            'note_update_method': 'append',
+            'custom_attributes': visitor.customAttributes
+          }
+        });
+      }, function(error) {
+        console.log('An error occurred: ', error);
+      }
+    );
+  }
+
+  updateVisitorInformation = function(attributeName, attributeValue){
+    getCurrentVisitor(salemove).then(function(visitor) {
+      visitor[attributeName] = attributeValue;
+      delete visitor.id;
+      delete visitor.href;
+      $.ajax({
+        type: "POST",
+        beforeSend: function (request)
+          {
+            requestHeaders = sm.salemoveApi.getRequestHeaders();
+              Object.keys(requestHeaders).forEach(function(key) {
+              request.setRequestHeader(key, requestHeaders[key]);
+            });
+          },
+        url: "https://api.salemove.com/visitor",
+        data: visitor
+        });
+      }, function(error) {
+        console.log('An error occurred: ', error);
+      }
+    );
+  }
+
```

Comment:
How does the above function negate the overwriting of previously stored information? 
If name: "james" 
and then this function is run with an alternative name, would it not still update it?